tools/flambda_backend_objinfo.ml

```diff
@@ -204,220 +213,209 @@ let print_cmxa_infos (lib : Cmx_format.library_infos) =
-    | Unexpected_error err -> explain_unexpected_error err)
-
-(* assume that 'ic' is already positioned at the right place
-   depending on the format (usually right after the magic number,
-   but Exec and Cmxs differ) *)
+  exit_magic_msg
+    Magic_number.(
+      match err with
+      | Parse_error err -> explain_parse_error expected_kind err
+      | Unexpected_error err -> explain_unexpected_error err)
+
+(* assume that 'ic' is already positioned at the right place depending on the
+   format (usually right after the magic number, but Exec and Cmxs differ) *)
 let dump_obj_by_kind filename ic obj_kind =
   let open Magic_number in
   match obj_kind with
-    | Cmo ->
-       let cu_pos = input_binary_int ic in
-       seek_in ic cu_pos;
-       let cu = input_value ic in
-       close_in ic;
-       print_cmo_infos cu
-    | Cma ->
-       let toc_pos = input_binary_int ic in
-       seek_in ic toc_pos;
-       let toc = (input_value ic : library) in
-       close_in ic;
-       print_cma_infos toc
-    | Cmi | Cmt ->
-       close_in ic;
-       let cmi, cmt = Cmt_format.read filename in
-       begin match cmi with
-         | None -> ()
-         | Some cmi ->
-            print_cmi_infos cmi.Cmi_format.cmi_name cmi.Cmi_format.cmi_crcs
-       end;
-       begin match cmt with
-         | None -> ()
-         | Some cmt -> print_cmt_infos cmt
-       end
-    | Cmx _config ->
-       let ui = (input_value ic : unit_infos) in
-       let crc = Digest.input ic in
-       close_in ic;
-       print_cmx_infos (ui, crc)
-    | Cmxa _config ->
-       let li = (input_value ic : library_infos) in
-       close_in ic;
-       print_cmxa_infos li
-    | Exec ->
-       (* no assumptions on [ic] position,
-          [dump_byte] will seek at the right place *)
-       dump_byte ic;
-       close_in ic
-    | Cmxs ->
-       (* we assume we are at the offset of the dynamic information,
-          as returned by [find_dyn_offset]. *)
-       let header = (input_value ic : dynheader) in
-       close_in ic;
-       print_cmxs_infos header;
-    | Ast_impl | Ast_intf ->
-       exit_errf "The object file type %S \
-                  is currently unsupported by this tool."
-         (human_name_of_kind obj_kind)
+  | Cmo ->
+    let cu_pos = input_binary_int ic in
+    seek_in ic cu_pos;
+    let cu = input_value ic in
+    close_in ic;
+    print_cmo_infos cu
+  | Cma ->
+    let toc_pos = input_binary_int ic in
+    seek_in ic toc_pos;
+    let toc = (input_value ic : library) in
+    close_in ic;
+    print_cma_infos toc
+  | Cmi | Cmt ->
+    close_in ic;
+    let cmi, cmt = Cmt_format.read filename in
+    begin
+      match cmi with
+      | None -> ()
+      | Some cmi ->
+        print_cmi_infos cmi.Cmi_format.cmi_name cmi.Cmi_format.cmi_crcs
+    end;
+    begin
+      match cmt with None -> () | Some cmt -> print_cmt_infos cmt
+    end
+  | Cmx _config ->
+    let ui = (input_value ic : unit_infos) in
```

Comment:
Shouldn't it be `unit_infos_raw`?

Reply:
https://github.com/ocaml-flambda/flambda-backend/pull/1029

Reply:
...which has been merged.  Have rebased this.